scripts/pipelines/gcloud/pipeline_generator.sh

```diff
@@ -42,17 +47,35 @@ red='\e[0;31m'
 
 # Common var
 commonTemplatesPath="scripts/pipelines/gcloud/templates/common" # Path for common files of the pipelines
+commonPipelineTemplatesPath="scripts/pipelines/common/templates/" # Path for common files of the pipelines
 pipelinePath=".pipelines" # Path to the pipelines.
 scriptFilePath=".pipelines/scripts" # Path to the scripts.
 export provider="gcloud"
 pipeline_type="pipeline"
+currentPath=`pwd`
+cd $localDirectory
+gitOriginUrl=$(git config --get remote.origin.url)
+export gCloudProject=$(echo "$gitOriginUrl" | cut -d'/' -f5)
+gCloudRepo=$(echo "$gitOriginUrl" | cut -d'/' -f7)
+cd $currentPath
 
 function obtainHangarPath {
 
     # This line goes to the script directory independent of wherever the user is and then jumps 3 directories back to get the path
     hangarPath=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && cd ../../.. && pwd )
 }
 
+function checkMachineType {
+
+    # The type of machine can only be two possible values:
+    if [[ "$machineType" != "E2_HIGHCPU_8" ]] && [[ "$machineType" != "E2_HIGHCPU_32" ]] && [[ "$machineType" != "N1_HIGHCPU_8" ]] && [[ "$machineType" != "N1_HIGHCPU_32" ]]
+    then
+      echo -e "${red}The machineType value is not correct, please between: 'E2_HIGHCPU_8', 'E2_HIGHCPU_32', 'N1_HIGHCPU_8' or 'N1_HIGHCPU_32'."
+      echo -e "For more info about the machineType: https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds?hl=fr#machinetype${white}"
```

Comment:
```suggestion
      echo -e "${red}Error: Chosen machine type is not a valid one." >&2
      echo -e "${red}Use -h or --help flag to display help." >&2
      echo -e "${red} Also check official documentation: https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds?hl=en#machinetype" >&2
      echo -ne ${white} >&2
```

Let's make it more maintainable.